app/(frontend)/(client-side)/product/[data]/page.tsx

```diff
@@ -0,0 +1,115 @@
+import ColorDisplay from "@/app/components/admin/ColorDisplay";
+
+const ProductDetailPage = ({ params }: { params: { data: string } }) => {
+  const [idSelected, setId] = useState<number>(0);
+  const [sizeSelected, setSizeId] = useState<number>();
+  const [quantity, setQuantity] = useState<number>(0);
+  const [maxQuantity, setMaxQuantity] = useState<number>(0)
+  const productInfo = JSON.parse(decodeURIComponent(params.data));
+
+  return (
+      <form className="grid grid-cols-1 md:grid-cols-2 p-4 md:p-8">
+        <section className="flex justify-center p-4 h-full w-full">
+          <Image
+              alt={productInfo.name}
+              src={productInfo.Attribute[idSelected ? idSelected : 0].image}
+              width={450}
+              height={450}
+              className="max-h-full w-auto"
+          />
+        </section>
+        <section className="font-normal p-4">
+          <header className="font-semibold text-xl">{productInfo.name}</header>
+          <p>{productInfo.price} BOB</p>
+          <section className="my-3">
+            <p className="mb-3">SIZE</p>
+            <div className="grid grid-cols-3 sm:grid-cols-6 sm:grid-rows-1 grid-flow-row gap-5">
+              {productInfo.Attribute.map((attribute: any, i: number) => {
+                return (
+                    <>
+                      {attribute.size.size_id === sizeSelected ? (
+                          <Button
+                              type="button"
+                              key={i}
+                              className="bg-orange-200 border border-black rounded-none"
+                              onClick={() => setSizeId(attribute.size.size_id)}
+                          >
+                            {attribute.size.name}
+                          </Button>
+                      ) : (
+                          <Button
+                              type="button"
+                              key={i}
+                              className="hover:bg-orange-200 border border-black rounded-none"
+                              onClick={() => setSizeId(attribute.size.size_id)}
+                          >
+                            {attribute.size.name}
+                          </Button>
+                      )}
+                    </>
+                );
+              })}
+            </div>
+          </section>
+          <section>
+            <p className="mb-3">COLOR</p>
+            <div className="grid sm:grid-cols-10 grid-cols-5 sm:grid-rows-1 grid-flow-row gap-2 place-content-center">
+              {productInfo.Attribute.map((attribute: any, i: number) => {
+                return (
+                    <Button
+                        type="button"
+                        key={i}
+                        className="bg-transparent items-center"
+                        onClick={() => setId(i)}
+                    >
+                      <ColorDisplay hexColor={attribute.color.name} diameter={30} />
```

Comment:
Put a margin in the colors, since being white it cannot be distinguished with the background, besides it has to have another background for the selected one.
![image](https://github.com/Los-creativos/cloteck-e/assets/100800348/a5422a2d-7d35-4194-993b-2bfdd2f97a53)